### PR TITLE
#120: API core view LayoutLeft transition

### PR DESCRIPTION
**Pull request by marcinwrobel1986**
* API/core/view/LayoutLeft transition from .md to .rst

before (.md) | after (.rst)
:----------------: | :-------------:
image | image

### CODE REVIEW

**Comment by cz4rs** [comment]
Note: this also enables building the documentation on pull requests (deployment only occurs on push to `main` as before).
image

**Review by cz4rs — APPROVED** [review]
I can approve Marcin's part in this (first commit). I need someone else to take a look at my changes.

File: docs/source/API/core/view/layoutLeft.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+==============
+
+.. role:: cppkokkoskokkos(code)
+   :language: cppkokkoskokkos
+
+Header File: ``Kokkos_Layout.hpp``
+
+Usage:
+
+.. code-block:: cppkokkos
```
Comment (cz4rs) [inline docs/source/API/core/view/layoutLeft.rst:11]: TODO: check if using `cppkokkos` (instead of `cpp`) makes sense here.

File: .github/workflows/deploy_docs.yml [hunk 2]
```diff
@@ -1,8 +1,9 @@
-name: Deploy docs
+name: Build and deploy docs
 
 on:
   push:
     branches: [ main ]
+  pull_request:
```
Comment (fnrizzi) [inline .github/workflows/deploy_docs.yml:6]: we should separate this into a different PR

Comment (cz4rs) [reply .github/workflows/deploy_docs.yml:6]: Posted #240.

File: docs/source/API/core/view/layoutLeft.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
+``LayoutLeft``
+==============
+
+.. role:: cppkokkoskokkos(code)
+   :language: cppkokkoskokkos
+
```
Comment (fnrizzi) [inline docs/source/API/core/view/layoutLeft.rst:6]: this has too many kokkoses in it... 
```
.. role:: cppkokkos(code)
   :language: cppkokkos
```

File: docs/source/API/core/view/layoutLeft.rst (hunk 1 above)
Comment (fnrizzi) [reply docs/source/API/core/view/layoutLeft.rst:11]: no, cppkokkos should NOT be used for code blocks

File: docs/source/API/core/view/layoutLeft.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+Usage:
+
+.. code-block:: cppkokkos
+
+   Kokkos::View<float*, Kokkos::LayoutLeft> my_view;
+
+Synopsis
+--------
+
+.. code-block:: cppkokkos
```
Comment (fnrizzi) [inline docs/source/API/core/view/layoutLeft.rst:18]: see cmment above

**Comment by fnrizzi** [comment]
@cz4rs can you please paste here the screen shot of what the new doc looks like?

**Comment by cz4rs** [comment]
Before:
image
image


After:
image
image

File: docs/source/API/core/view/layoutLeft.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    Default copy constructor, element-wise copies the other Layout
+
+  .. cppkokkos:function:: LayoutLeft(LayoutLeft&&)
+
+    Default move constructor, element-wise moves the other Layout
+
+  .. code-block:: cpp
+
+     KOKKOS_INLINE_FUNCTION
+     explicit constexpr LayoutLeft(size_t N0 = 0, size_t N1 = 0, size_t N2 = 0,
```
[the remark this answers is not in the document]
Comment (cz4rs) [reply docs/source/API/core/view/layoutLeft.rst]: Something's not quite right:
image

[the remark this answers is not in the document]
Comment (cz4rs) [reply docs/source/API/core/view/layoutLeft.rst]:
```
reading sources... [100%] API/core/view/layoutLeft                                                                                                                      
/home/cz4rs/Kokkos/kokkos-core-wiki/docs/source/API/core/view/layoutLeft.rst:74: WARNING: Error when parsing kokkosinlinefunction declaration.
If the function has no return type:
  Error in declarator or parameters-and-qualifiers
  Invalid C++ declaration: Expected identifier in nested name, got keyword: explicit [error at 8]
    explicit constexpr LayoutLeft(size_t N0 = 0, size_t N1 = 0, size_t N2 = 0, size_t N3 = 0, size_t N4 = 0, size_t N5 = 0, size_t N6 = 0, size_t N7 = 0);
    --------^
If the function has a return type:
  Invalid C++ declaration: Expected identifier in nested name, got keyword: explicit [error at 8]
    explicit constexpr LayoutLeft(size_t N0 = 0, size_t N1 = 0, size_t N2 = 0, size_t N3 = 0, size_t N4 = 0, size_t N5 = 0, size_t N6 = 0, size_t N7 = 0);
    --------^
```

**Review by ldh4 — APPROVED** [review]
